### ISSUE TITLE
Greenhouse: Allow user to select which endpoints to sync when creating the source

### ISSUE THREAD

**Issue by kyle-cheung** [airbytehq/airbyte#5343]
## Tell us about the problem you're trying to solve
I'm trying to setup a Greenhouse source for only a handful of endpoints, however, I'm running into the following error:
```{"message":"This API Key does not have permission for this endpoint"}```

I suspect this is because Airbyte (and this is not listed in the documentation) requires the user's API key to have access to ALL or specific Greenhouse endpoints. We do not want to bring all in due to PII concerns and simply there are a handful of endpoints that is currently of no use. There is no way to get pass creating a source connector without enabling full access to all Greenhouse endpoints from which to then select which schemas to load on the connections page.

tldr: I get stuck on creating a source because I have limited access to our Greenhouse endpoints and when setting up a source I cannot select which endpoints I have access to so it throws a permissions error.

## Describe the solution you’d like
Allow the user to select which endpoints to sync when creating the source so we can set our access accordingly

## Describe the alternative you’ve considered or used
N/A as we do not want to give more access to the API key

## Are you willing to submit a PR?
Not technical enough!

**Comment by midavadim** [comment]
@kyle-cheung could you please provide us with the list of streams/endpoints for which you have access

**Comment by kyle-cheung** [comment]
@midavadim 
I have access to these ones:
- close reasons
- demographic data
- departments
- job openings
- job posts
- job stages
- jobs
- offers
- rejection reasons
- scheduled interviews
- sources
I'm not sure what the specific permissions are but my API key has been granted access to read those endpoints

**Comment by midavadim** [comment]
![greenhouse](https://user-images.githubusercontent.com/3889748/133885288-b090db00-3dca-42c4-a78a-f4531d99f804.png)

API key can be granted with permissions to any combinations of endpoints. 
User who uses that API key might don't know a list of endpoints he has access to.

Solution: programmatically try to read each supported stream and process only 'accessible' streams during connector setup and sync
